mne/io/nirx/nirx.py

```diff
@@ -19,15 +19,25 @@
 
 
 @fill_doc
-def read_raw_nirx(fname, preload=False, verbose=None):
+def read_raw_nirx(fname, nosatflags=False, preload=False, verbose=None):
     """Reader for a NIRX fNIRS recording.
 
-    This function has only been tested with NIRScout devices.
+    This function has only been tested with NIRScout and NIRSport1 devices.
 
     Parameters
     ----------
     fname : str
         Path to the NIRX data folder or header file.
+    nosatflags : bool
```

Comment:
Can we try to find a better parameter name? Something without any abbreviations, such as `saturated=False` or `flag_saturated=False` would be much better IMO.

Reply:
I figured I could use the name of the file, but your request does make sense. Maybe `use_saturated` ? To be more specific than just `saturated`. @rob-luke, any idea ?

Reply:
I think we need to establish the preferred behaviour of the function first, then answer these detail questions. But my vote is for `return_saturated` or `use_saturated`.

Reply:
IMO `flag_saturated` is clearer, because it implies that saturated values are flagged somehow (with NaNs in this case). Even if we don't flag/mask saturated values, we will always return saturated values (either masked/flagged or not), so `use_saturated` or `return_saturated` does not tell me what they do. So I'd vote for either `flag_saturated` or `mask_saturated`, both of which return NaNs for saturated values if `True` and the raw values if `False`.

Reply:
Lets go with `mask_saturated` then.

Reply:
Just to be clear : `mask_saturated` is set to **True** when we want `NaN` values, right ?

Reply:
What does `False` do? Maybe it's clearest just to name the behaviors like `saturated='nan' | 'whatevertheotherbehavioris'`.

Reply:
Currently with `nosatflags`, `True` means we use the *.nosatflags_wlX files, whereas `False` means we use the standard *.wlX files (which contains 'NaN' values)

Reply:
`mask_saturated=True` masks saturated values with NaNs, `mask_saturated=False` doesn't mask saturated values.

Reply:
If you want `saturated='nan'`, the other alternative should be "do nothing", so maybe `saturated=None`? Not sure if this is more intuitive though.

Reply:
`saturated='ignore'` would work, too, but I'm also okay with `mask_saturated` as a bool.

Reply:
Yes, I like that! Makes for a shorter name too.

Reply:
If there will only ever be 2 options, please keep as bool. `saturated='nan'` or `'ignore'` leads to internal code that is uglier and more annoying to maintain IMO, and the API is just as clear I think.

Reply:
I'm thinking exactly of leaving the possibility open for future behaviors like `saturated='annotate'`, which could for example annotate as `BAD` segments of saturation like `mne.preprocessing.mark_flat` does...

Reply:
OK,that makes sense. I was envisioning the simple "do x" vs "don't do x" case, where I think Boolean has obvious advantages.

Reply:
> If there will only ever be 2 options ...

I think this is the more important thing to be discussed first. What should be returned by default, and what should the other options be. Its unclear to me what the correct behaviour should be. And I still like the idea of returning a mask as well as the data somehow, but maybe this is too complicated as discussed earlier.

Reply:
I thought that this was already settled. `saturated='annotate'` returns signals with NaNs, whereas `saturated='ignore'` returns the actual values.

Reply:
And by default I would use `saturated='ignore'` (so no NaNs).

Reply:
My suggestion was saturated='nan' populates with Nan, saturated='annotate' would actually add raw annotations like mark_flat (but we can add this later, just Nan and ignore to start)

Reply:
I like this last suggestion by larsoner. Personally I would imagine myself using the annotate option, this way I could use all filtering functions without caring about NaNs. But then easily reject epochs or data segments where I know bad data lives. Are there any negatives to this approach? Also saturated=nan is intuitive to what is going to happen to the saturated data. We could also keep the saturated=ignore option which returns the data but doesn't add annotations if that's what people want.

Reply:
Filtering should work in the presence of NaNs (at least `scipy.signal.lfilter` has no problems with NaNs).

Reply:
Changing 'mask' with 'nan' then ?

Reply:
> Filtering should work in the presence of NaNs

Nice, I didn't know this. But I guess more generally I don't write my functions expecting NaNs, perhaps this is something I should think of more in the future.

My vote is for larsoners proposal above.

Reply:
My bad, I actually always agreed with @larsoner but I wrote `'annotate'` instead of `'nan'`. So yes, I think we now all agree on the solution:

- `saturated='ignore'` (the default) returns the original values (contained in .nosatflags_wl* files)
- `saturated='nan'` replaces saturated values with NaNs (contained in .wl* files)
- `saturated='annotate'` will return the original values (just like `'ignore'`) and in addition will create annotations at the locations of saturated values - however, this option is for another PR

Reply:
Ok we are all on the same page now. Thanks for clarifying cbrnr.

> however, this option is for another PR

Can we do this all in this PR? There is no rush to get this merged as we don't even have a test set committed to the repo yet. I don't want to have incomplete support for NIRSport merged, then we are just going to have triage more issues. Let's add complete support in one hit. If annotate is too much work for one person, then multiple people can commit to this branch.

Reply:
Also thanks for bearing with us Swy7ch. Let us know if there are any other decisions to be made that are blocking your progress.

Reply:
Don't worry, I actually think the process is quite active, which is a good think as far as I'm concerned :)

>Can we do this all in this PR?

Definitely ! I guess you already suggested it #7926, so I didn't ask again. I'll see what I can do concerning the annotation. A few questions in this regard:

* the data should be used as is (i.e. with NaN values), but the channels must be marked(like the 'bad' channels markin the [tutorial](https://mne.tools/dev/auto_tutorials/preprocessing/plot_70_fnirs_processing.html#evaluating-the-quality-of-the-data)), right ? I don't fully understand @drammock's [solution](https://github.com/mne-tools/mne-python/pull/7936#issuecomment-650445349)
* any idea of the name of the marking ? 'saturated', I assume ?
* do you know if marking an object twice could lead to troubles ? I'll try to find a dataset with channels that are both of bad quality and containing NaN anyway, it's just to have a heads-up.

Reply:
> the data should be used as is (i.e. with NaN values), but the channels must be marked (like the 'bad' channels markin the tutorial), right ? I don't fully understand @drammock's solution

TL;DR: you don't need to understand my solution, because it looks like we're not going to do that anyway (which is fine).

Full explanation: My suggestion of using `numpy.ma.masked_array` was an alternative to doing separate masked / non-masked options in the reader function. In a masked array approach, the data includes all the saturated values, but the nan-masked values are also read from disk and used to create a numpy mask over the data array. This is nice because it gives users the option to change their mind about using/not using the saturated values later on (without having to reload from disk) because masked arrays make it easy to ignore the mask and see the underlying data if you want to. But the downside is that most of the downstream code (analysis and plotting functions that operate on `Raw` objects) is not tested to work with masked arrays and I expect several things will break. The good news is, by having string options like `ignore` and `nan` and `annotate`, it leaves open the possibility to add a `masked` option later if we find that we need/want Raw objects to support masked arrays for other reasons.